phpBB/includes/functions.php

```diff
@@ -2745,7 +2745,8 @@ function get_preg_expression($mode)
 			return array(
 				'#<!\-\- e \-\-><a href="mailto:(.*?)">.*?</a><!\-\- e \-\->#',
 				'#<!\-\- l \-\-><a (?:class="[\w-]+" )?href="(.*?)(?:(&amp;|\?)sid=[0-9a-f]{32})?">.*?</a><!\-\- l \-\->#',
-				'#<!\-\- ([mw]) \-\-><a (?:class="[\w-]+" )?href="(.*?)">(.*?)</a><!\-\- \1 \-\->#',
+				'#<!\-\- ([mw]) \-\-><a (?:class="[\w-]+" )?href="http://(.*?)">\2</a><!\-\- \1 \-\->#',
```

Comment:
I realy don't get the point why to hardcode a http:// prefix at all? Especially as the world is switching from http:// to https:// for good and nowadays http(S):// is not the only prefixes out there.
Why not getting rid of any hardcoded prefix?
And why prefixing anything with http:// when you don't know if it's intended to be http:// and not https:// or something else or even realy nothing?
This leads to unexpected behavior and false positives.

When i post www.phpbb.com which prefix did i mean? http or https or ftp or git or ssh or imap or foo?